x-pack/plugins/security_solution/public/common/hooks/use_app_toasts.ts

```diff
@@ -72,13 +81,152 @@ type MaybeESError = IEsError & { err?: Record<string, unknown> };
+ * REST API's that are typically of a particular format and form.
+ *
+ * The existing error_toaster code tries to consolidate network and software stack traces but really
+ * here and our toasters we are using them for network response errors so we can troubleshoot things
+ * as quick as possible.
+ *
+ * We override and use error.stack to be able to give _full_ network responses regardless of if they
+ * are from Kibana or if they are from elasticSearch since sometimes Kibana errors might wrap the errors.
+ *
+ * Sometimes the errors are wrapped from io-ts, Kibana Schema or something else and we want to show
+ * as full error messages as we can.
+ */
+export const appErrorToErrorStack = (error: AppError): Error => {
+  const statusCode = isKibanaError(error)
+    ? `(${error.body.statusCode})`
+    : isSecurityAppError(error)
+    ? `(${error.body.status_code})`
+    : '';
+  const stringifiedError = getStringifiedStack(error);
+  const adaptedError = new Error(
+    `${String(error.body.message).trim() !== '' ? error.body.message : error.message} ${statusCode}`
+  );
+  // Note although all the Typescript typings say that error.name is a string and exists, we still can encounter an undefined so we
+  // do an extra guard here and default to empty string if it is undefined
+  adaptedError.name = error.name != null ? error.name : '';
+  if (stringifiedError != null) {
+    adaptedError.stack = stringifiedError;
+  }
+  return adaptedError;
+};
+
+/**
+ * Takes an error and tries to stringify it and use that as the stack for the error toaster
+ * @param error The error to convert into a message
+ * @returns The exception error to return back
+ */
+export const errorToErrorStack = (error: Error): Error => {
+  const stringifiedError = getStringifiedStack(error);
+  const adaptedError = new Error(error.message);
+  adaptedError.name = error.name;
+  if (stringifiedError != null) {
+    adaptedError.stack = stringifiedError;
+  }
+  return adaptedError;
+};
+
+/**
+ * Last ditch effort to take something unknown which could be a string, number,
+ * anything. This usually should not be called but just in case we do try our
+ * best to stringify it and give a message, name, and replace the stack of it.
+ * @param error The unknown error to convert into a message
+ * @returns The exception error to return back
+ */
+export const unknownToErrorStack = (error: unknown): Error => {
+  const stringifiedError = getStringifiedStack(error);
+  const message = isString(error)
+    ? error
+    : error instanceof Object && stringifiedError != null
+    ? stringifiedError
+    : String(error);
+  const adaptedError = new Error(message);
+  adaptedError.name = message;
+  if (stringifiedError != null) {
+    adaptedError.stack = stringifiedError;
+  }
+  return adaptedError;
+};
+
+/**
+ * Stringifies the error. However, since Errors can string into empty objects this will
```

Comment:
`Errors can string`? Can turn strings into empty objects?

Reply:
Uh, yeah, let me fix that bad English. I meant errors can JSON.stringify() into empty objects since their properties are not enumerable.